### ISSUE TITLE
Add option to resolve ARK references, from Archive.org etc. simliar to DOI

### ISSUE THREAD

**Issue by wujastyk** [JabRef/jabref#6696]
Archive.org auto-assigns an [Archival Resource Key](https://en.wikipedia.org/wiki/Archival_Resource_Key) (ARK) to every item in its collection.  ARKs are used by other institutions too.  They function pretty much in the same way as DOIs.  The main resolver for ARKs is [http://n2t.net](http://n2t.net).  They are designed to be persistent URIs.  Many people use the ordinary URL of items in Archive.org for reference, but these URLs are messy, often very long, and assigned randomly by individual uploaders on no uniform scheme.

At present, I enter an ark into the Eprint field, and an Eprinttype of "ark".  E.g.,

```
@Book{kale-1928,
  author      = {M. R. Kale},
  title       = {Bāṇa's Kādambarī [Pūrvabhāga Complete]},
  location    = {Delhi etc.},
  publisher   = {Motilal Banarsidass},
  date        = {1928},
  edition     = {3},
  eprint      = {ark:/13960/t8cg89f76},
  eprinttype = {ark},
}
```

In my LaTeX style file, I have this Biblatex macro: 
```
\DeclareFieldFormat{eprint:ark}{%
	\textsc{ark:}\space
	\ifhyperref  
	{\href{https://n2t.net/#1}{\nolinkurl{#1}}}
	{\nolinkurl{#1}}}
```

This produces satisfactory results in bibliographies. 

But in JabRef, I would like to be able to click a link and be taken to [http://n2t.net/ark/13960/t8cg89f76](http://n2t.net/ark/13960/t8cg89f76), just as I can with a DOI or URL.  

Is there a secret way to achieve this?  To get a link into the Linked Identifiers column?  If not, might a feature be added to JabRef to make this possible?  It could be something hard-wired just for ARKs, or it could be a rule-based system that interpreted eprint and eprinttype data according to a rule like \DeclareFieldFormat above.

**Comment by akwala** [comment]
I'd like to take this on. I've read the Contribution document. Will touch base shortly in your chat channel.

**Comment by akwala** [comment]
Drafting the spec for this enhancement... Please provide feedback and tips, useful info, etc. are also welcome. (I'm just beginning to look at the code.)

### Implement a new `ID Type`, _ARK_, that is similar to _DOI_

- Add _ARK_ to the "General" tab in the entry detail window with the same buttons/functionality as _DOI_ – use http://n2t.net/ to resolve an _ARK_.
![Screenshot_20201011_200058_scaled](https://user-images.githubusercontent.com/1786/95694649-2b8b9280-0c01-11eb-9031-de84cbacb420.png)

- Add _ARK_ to the `ID Type` list (dropdown) in the "New entry" window and mimic the options/actions for _DOI_.
![Screenshot_20201011_200750](https://user-images.githubusercontent.com/1786/95694980-d05a9f80-0c02-11eb-9722-fa168c2be99a.png)

**Comment by wujastyk** [comment]
Yes, parallel treatment to DOI for lookup and resolution seems the right
way, as far as I can see.

**Comment by Siedlerchr** [comment]
Hi, ARK is not a fetcher as far as I see there no way to get bibtex from an ark link.
I would just implement it in the entry editor view model/field  for the eprint type. Similar to Arxiv maybe.
I vaguely remember there is some handling for the eprint field

**Comment by koppor** [comment]
JabRef needs to construct the BibTeX. It needs to apply some heuristics. For the example, the BibTeX is only

```bibtex
@misc{key,
    eprint      = {ark:/13960/t8cg89f76},
  eprinttype = {ark},
}
```

There is a side issue: We support `doi` as field (`doi = {10.1007/s00450-019-00407-8}`), but `eprint = {doi:10.1007/s00450-019-00407-8}, eprinttype = {doi}` is the same.

@wujastyk I wonder why `ark` appears both in `eprint` and `eprinttype`. Isn't the prefix `ark:` enough?

**Comment by akwala** [comment]
> There is a side issue: We support `doi` as field (`doi = {10.1007/s00450-019-00407-8}`), but `eprint = {doi:10.1007/s00450-019-00407-8}, eprinttype = {doi}` is the same.

By "issue" do you mean this shouldn't be the case?

I wondered whether the "ark:/" should be in the `eprint` field. Is there a convention or rule for what comprises an `eprint` value?

(I am new to BibTeX.)

**Comment by wujastyk** [comment]
I have never seen an ARK reference without the ark:/ prefix.  So I have
presumed that it is not to be split off.  But I could be wrong.

There's a statement of the standard here:

   - https://en.wikipedia.org/wiki/Archival_Resource_Key

that says the structure is:

[http://NMAH/]ark:/NAAN/Name[Qualifier]
The NMAH, Name Mapping Authority Host, is the resolver (commonly
http://n2t.net).  But it does look to me as if the "ark:/" string and the
NAAN are always present.

**Comment by wujastyk** [comment]
I understand now that ARK is not a system for fetching BibTeX data (like DOI).  

But I still think it should not be impossible to do something less ambitious in JabRef in order to get an ARK link appearing and live in the Linked Identifiers column of the database display.  (I.e., putting the resolver URL, http://n2t.net on the front of the data from the eprint field.)

I think entering ark references using eprint and eprinttype is fine from the user and BibTeX point of view.  But if having an actual ark field would materially help with getting the data into the Linked Identifiers column, then that should be no problem either.

So, I would prefer for this issue to remain open for the sake of this more modest goal of getting a clickable link into the Linked Identifiers column.

**Comment by wujastyk** [comment]
**[Feature suggestion]**

JabRef 5.5--2021-12-28--8e92228
Linux 5.13.0-22-generic amd64 
Java 16.0.2 
JavaFX 17.0.1+1

+1 

For constructing clickable ARK refs in the "linked files" column, as mentioned [above](https://github.com/JabRef/jabref/issues/6696#issuecomment-818097539).  

These could be constructed out of the `eprint` and `eprinttype` fields thus:  

If `eprrinttype` has value `ark` then the linked files column can have a link to `http://n2t.net/<eprint value>`.

This could become a small, more general subsystem.  JabRef could have an option to define a rule for any such linkable external source.  In an options menu one could specify "if `eprinttype` = X, put link `X-prefix/<eprint value>` into the linked files column. Just a thought.   

In the back of my mind, I'm thinking of BibLaTeX macros like this that I use routinely (BibLaTeX manual, 4.11.2):

```
\DeclareFieldFormat{eprint:ark}{%
	\textsc{ark:}\space
	\ifhyperref  
	{\href{https://n2t.net/#1}{\nolinkurl{#1}}}
	{\nolinkurl{#1}}}

\DeclareFieldFormat{eprint:academia}{%
	Academia.edu\addcolon\space
	\ifhyperref
	{\href{http://www.academia.edu/#1}{\nolinkurl{#1}}}
	{\nolinkurl{#1}}}
\DeclareFieldAlias{eprint:academia.edu}{eprint:academia}

\DeclareFieldFormat{eprint:tbrc}{%
    \textsc{TBRC}\space
    \ifhyperref  
    {\href{https://www.tbrc.org/\#!rid=#1}{\nolinkurl{#1}}}
	{\nolinkurl{#1}}}
```

**Comment by wujastyk** [comment]
Could this be considered for v.6?

**Comment by ThiloteE** [comment]
Related issue: https://github.com/JabRef/jabref/issues/6627#issuecomment-1002815225

**Comment by wujastyk** [comment]
_Archival Resource Key (ARK)_

I have been using this feature almost daily for a couple of years, and it works very well.  `eprint={ark:/fubar}, eprinttype={ark}`.

The resolver that we discussed above is `http://n2t.net`.  And it's fine.  But I have noticed other general resolvers out there, like `http://gallica.bnf.fr` etc.  As far as I can tell, the n2t.net resolver is hardwired into JabRef.  Would it be possible to make this configurable?  It's not essential, but seems not unreasonable.
